3scale/porta
# THREESCALE-7132: Add support for AWS STS authentication

**What this PR does / why we need it**:

This PR implements STS credentials that can be used to access S3 buckets instead of the IAM credentials used by default.

**Which issue(s) this PR fixes** 

Implements https://issues.redhat.com/browse/THREESCALE-7132.

**Some context first**
- With the current IAM-related ENV variables (`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`), calls to AWS should still work; 
- If the new STS-related ENV variables (`AWS_ROLE_ARN` and `AWS_WEB_IDENTITY_TOKEN_FILE`) are **also** present, we'll still call AWS with the IAM credentials;

**Verification steps** 
- With IAM credentials
  - Assert that the files we store on S3 are still available;
    - Examples of files we store on S3: `Invoice#pdf`, `Profile#logo`, `File#attachment`;
- With both IAM and STS credentials;
  - Assert that IAM credentials are used instead of the STS ones;
- With STS credentials (the new stuff)
  - Locally:
    - Have a `token` file somewhere in your local machine;
    - Set the `AWS_WEB_IDENTITY_TOKEN_FILE` so it points to the `token` file;
    - Set a valid `AWS_ROLE_ARN`;
    - Assert that the files we store on S3 are still accessible;
  - RHOAM:
    - The `token` file should already be available from an Operator;
    - Assert that the ENV variables are set;
    - Assert that the files we store on S3 are still accessible;
